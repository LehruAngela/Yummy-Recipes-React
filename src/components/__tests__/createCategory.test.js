import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CreateCategory from '../createCategory';

describe('CreateCategory', () => {
    const wrapper = shallow(<CreateCategory/>);
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        }); 
    
    it('handles CreateCategory', () => {
        expect(wrapper.instance().handleCreateCategory({ preventDefault }));
        });


})