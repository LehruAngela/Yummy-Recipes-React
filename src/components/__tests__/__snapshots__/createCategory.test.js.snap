// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCategory renders properly 1`] = `
<div>
  <button
    className="btn my-2 my-sm-0 addbutton link"
    data-target="#createCategory"
    data-toggle="modal"
    type="button"
  >
    <i
      class="fas fa-plus-circle"
    />
     Add Category
  </button>
  <div
    aria-hidden="true"
    aria-labelledby="exampleModalCenterTitle"
    class="modal fade"
    id="createCategory"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h4
            class="modal-title"
            id="exampleModalLongTitle"
          >
            Add Category
          </h4>
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            id="closeCategoryModal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <h6>
                <label>
                  Category Name
                </label>
                <br />
              </h6>
              <input
                name="category_name"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <button
                type="submit"
              >
                Create
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
